### PR TITLE
Performance/878 zoom performance

### PR DESCRIPTION
**Pull request by Nova-2119**
### Benefits
Improved scroll responsiveness

### Possible Drawbacks
Unknown

### Applicable Issues

https://github.com/constellation-app/constellation/issues/878

### CODE REVIEW

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java [hunk 1]
```diff
@@ -269,46 +269,22 @@ public Vector3f windowToGraphCoordinates(final Camera camera, final Point point)
...
-                    closestDistance = closestNode.getLength();
-                }
-            }
-        }
-        return closestNode;
+        Stream<NodeCameraDistance> nodeCameraDistances = VisualGraphUtilities.streamVertexSceneLocations(graph, camera)
+                                                .parallel()
+                                                .map(vector -> new NodeCameraDistance(vector, horizontalScale, verticalScale));
+                                                
+        NodeCameraDistance closest = nodeCameraDistances.parallel().reduce(new NodeCameraDistance(), (ncd1, ncd2) -> NodeCameraDistance.getClosestNode(ncd1, ncd2));
```
Comment (antares1470) [inline CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java:285]: These 2 can be final

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java [hunk 2]
```diff
@@ -335,4 +311,71 @@ public float getDPIScalingFactor() {
             return 1.0f;
         }
     }
+    
+        
+    private static class NodeCameraDistance{
+        Vector3f nodeLocation;
+        Float distanceFromCamera;
```
Comment (antares1470) [inline CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java:318]: These look like they could be final

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java (hunk 1 above)
Comment (arcturus2) [reply CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java]: Updated

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java (hunk 2 above)
Comment (arcturus2) [reply CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/InteractiveGLVisualProcessor.java]: Updated

**Pull request merged**